src/crypto/chacha20.h

```diff
@@ -20,7 +20,9 @@ class ChaCha20
     void SetKey(const unsigned char* key, size_t keylen);
     void SetIV(uint64_t iv);
     void Seek(uint64_t pos);
-    void Output(unsigned char* output, size_t bytes);
+
+    void Output(unsigned char* c, size_t bytes);
+    void Crypt(const unsigned char* input, unsigned char* output, size_t bytes);
```

Comment:
Now that this class can encrypt plaintext, the class comment should be updated from "A PRNG class for ChaCha20."

Really, I'd like to see more commenting for this class in general (ie note that `SetIV` sets the nonce, `Seek` sets the counter, that the counter is 64 bytes and the nonce is 64 bytes as in the original ChaCha20 spec instead of 32 byte counter/96 byte nonce as in the IETF spec).

(there are a couple of good resources on the different counter/nonce sizes here: https://libsodium.gitbook.io/doc/advanced/stream_ciphers/chacha20 and here: http://loup-vaillant.fr/tutorials/chacha20-design)

Reply:
The IETF has a slightly modified recommendation AFAIK.
We use Bernstein-Vanilly 64bit nonce and 64bit block counter.

* Changed the class comment
* Added some function comments